verification.go

```diff
@@ -85,69 +85,10 @@ func VerifyBasicBlockFilter(filter *gcs.Filter, block *btcutil.Block) (int,
 			}
 		}
 
-		// Now we can go through all inputs and check that the filter
-		// also included any pk scripts of the outputs being _spent_.
-		// We can do this for witness items since the witness always
-		// contains the full script as the last element on the stack.
-		for inIdx, in := range tx.MsgTx().TxIn {
-			// There are too many edge cases to cover for non-
-			// witness scripts. And in LN land we're interested in
-			// witness spends only anyway. Therefore let's skip any
-			// input that has no witness.
-			//
-			// TODO(guggero): Add all those edge cases to
-			// ComputePkScript?
-			if len(in.Witness) == 0 {
-				continue
-			}
-
-			// The only input type that has both set is a nested
-			// P2PKH (P2SH-P2WKH). We can verify that one because
-			// the script hash has to be HASH160(OP_PUSH32 <PKH>).
-			script, err := txscript.ComputePkScript(
-				in.SignatureScript, in.Witness,
-			)
-
-			// Just skip any inputs that we can't derive the pk
-			// script from.
-			if err == txscript.ErrUnsupportedScriptType {
-				log.Tracef("Skipping filter validation for "+
-					"input %d of tx %v in block %v "+
-					"because script type is not supported "+
-					"for validating against filter", inIdx,
-					tx.Hash(), block.Hash())
-
```

Comment:
What if we leave most of this in tact (since the logging is very useful) and simply don't error out? Feels like there's a way to retain _most_ of this and validate what we can, but still need to meditate on it a bit more.